python-spec/src/somacore/query/query.py

```diff
@@ -305,8 +333,12 @@ def _read(
         *,
         column_names: AxisColumnNames,
         X_layers: Sequence[str],
+        obsm_keys: Sequence[str] = (),
+        obsp_keys: Sequence[str] = (),
+        varm_keys: Sequence[str] = (),
+        varp_keys: Sequence[str] = (),
     ) -> "_AxisQueryResult":
-        """Reads the entire query result into in-memory Arrow tables.
+        """Reads the entire query result in memory.
```

Comment:
we still need to sort this out, as it is a hodge podge of formats.   If it is only for use in the anndata code path, it really should return Pandas Dataframes for obs/var, etc.

If it is for other uses, it should return chunked arrays, not ndarray for obsm, etc.  (ie.., avoid doing the extra copy that ndarray implies).

Either way, lets pick a purpose for this routine and have it fulfill that destiny.  At the moment, it is a mish-mash.

This includes both this routine and its return type.

I'm OK with either approach, but note that Arrow-native types are more useful, and give the caller the choice of doing zero-copy or not (the current code forces a copy in some cases)

Reply:
At this stage, given that `_read` is a private API and is only used by `to_anndata`, I am proposing that we just merge the two methods. If in the future we'll introduce support for some other data format that can feed directly from Arrow objects, we can always go back and add back a properly defined `_read` method that is more general, and move the conversions to pandas/numpy/scipy back to the `to_anndata`. Alternatively we can just keep the two methods separate and push the conversion to pandas Dataframes up to `_read`.

Reply:
I don't have an opinion on the proposed merge (other than to note that giant methods are hard to follow).  I'm just saying that if it is for to_anndata, you might as well return AnnData-compat types (e.g., Pandas), etc.

this is a minor point - just seems like we have a half-modified API signature, which is worth cleaning up.

Reply:
I ended up making sure that the `_AxisQueryResult` object only has "final" types (i.e. dataframes, ndarray, scipy.sparse) and fixed the docstrings a bit. Let me know if we want to make the fact that this is - effectively - a proxy to an AnnData object more clear in the docs.